### PR TITLE
Add lint rule warning when just a namespace is exported

### PR DESCRIPTION
**Pull request by andy-ms**

### CODE REVIEW

File: tslint-definitions.json [hunk 1]
```diff
@@ -14,7 +14,6 @@
 		"no-reference": false, // TODO: forbid "../" reference, but not "./" reference
 		"no-shadowed-variable": false,
 		"no-string-literal": false,
-		"no-unused-variable": false,
```
Comment (andy-ms) [inline tslint-definitions.json]: Having this line caused tslint to warn that it was deprecated.

File: src/tslint/exportJustNamespaceRule.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import * as Lint from "tslint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+	static metadata: Lint.IRuleMetadata = {
+		ruleName: "export-just-namespace",
+		description: "Forbid to `export = foo` where `foo` is a namespace an isn't merged with a function/class.",
```
Comment (mhegazy) [inline src/tslint/exportJustNamespaceRule.ts]: typo, `and`

File: src/tslint/exportJustNamespaceRule.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	static FAILURE_STRING = "Instead of `export =`-ing a namespace, use the body of the namespace as the module body.";
+
+	apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+		return this.applyWithWalker(new Walker(sourceFile, this.getOptions()));
+	}
+}
+
+class Walker extends Lint.RuleWalker {
+	visitSourceFile(node: ts.SourceFile) {
+		const exportEqualsNode = node.statements.find(s => s.kind === ts.SyntaxKind.ExportAssignment) as ts.ExportAssignment | undefined;
```
Comment (mhegazy) [inline src/tslint/exportJustNamespaceRule.ts]: `ExportAssignment` is both `export = ` and `export default`. check `node.isExportEquals` as well.

File: src/tslint/exportJustNamespaceRule.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+			case ts.SyntaxKind.ModuleDeclaration:
+				anyNamespace = anyNamespace || nameMatches((statement as ts.ModuleDeclaration).name);
+				break;
+			case ts.SyntaxKind.VariableStatement:
+				if ((statement as ts.VariableStatement).declarationList.declarations.some(d => nameMatches(d.name))) {
+					// OK. It's merged with a variable.
+					return false;
+				}
+			case ts.SyntaxKind.FunctionDeclaration:
+			case ts.SyntaxKind.ClassDeclaration:
```
Comment (mhegazy) [inline src/tslint/exportJustNamespaceRule.ts]: Also `TypeAliasDeclaration` and `InterfaceDeclaration` for good measures.

**Pull request merged**